### PR TITLE
feat: enhance lsp semantic highlighting

### PR DESCRIPTION
**Pull request by mlwarner**
Uncomment lines for lsp semantic highlighting. This helps with treesitter highlight consistency as the lsp comes online.

I came across this while I was trying to understand color shifts and flickering during startup. My personal config contains these lines:

```lua
vim.api.nvim_set_hl(0, "@lsp.type.variable", { link = '@variable' })
vim.api.nvim_set_hl(0, "@lsp.type.property", { link = '@property' })
```

Can't comment on the other highlight groups, but the names are consistent.

#### Before

Highlight groups link to `Identifer` e.g. (@lsp.type.variable.lua links to Identifier priority: 125)

#### After

Highlight groups link to matching treesitter group e.g. (@lsp.type.variable.lua links to TSVariable priority: 125)

### CODE REVIEW

File: lua/base16-colorscheme.lua [hunk 1]
```diff
@@ -593,21 +593,20 @@ function M.setup(colors, config)
...
-        -- hi['@lsp.type.function'] = '@function'
-        -- hi['@lsp.type.method'] = '@method'
-        -- hi['@lsp.type.macro'] = '@macro'
-        -- hi['@lsp.type.decorator'] = '@function'
+        hi['@lsp.type.namespace'] = '@namespace'
+        hi['@lsp.type.type'] = '@type'
+        hi['@lsp.type.class'] = '@type'
+        hi['@lsp.type.enum'] = '@type'
+        hi['@lsp.type.interface'] = '@type'
+        hi['@lsp.type.struct'] = '@structure'
```
Comment (RRethy) [inline lua/base16-colorscheme.lua:601]: `@structure` doesn't exist.

File: lua/base16-colorscheme.lua [hunk 2]
```diff
@@ -593,21 +593,20 @@ function M.setup(colors, config)
...
+        hi['@lsp.type.parameter'] = '@parameter'
+        hi['@lsp.type.variable'] = '@variable'
+        hi['@lsp.type.property'] = '@property'
+        hi['@lsp.type.enumMember'] = '@constant'
+        hi['@lsp.type.function'] = '@function'
+        hi['@lsp.type.method'] = '@method'
+        hi['@lsp.type.macro'] = '@macro'
```
Comment (RRethy) [inline lua/base16-colorscheme.lua:608]: `@macro` doesn't exist.

File: lua/base16-colorscheme.lua (hunk 1 above)
Comment (mlwarner) [reply lua/base16-colorscheme.lua]: changed to `@type`

File: lua/base16-colorscheme.lua (hunk 2 above)
Comment (mlwarner) [reply lua/base16-colorscheme.lua]: Used `@function.macro`. Seems more appropriate than `@constant.macro`

**Comment by donovanglover** [comment]
FWIW this change caused functions to no longer have the function color when used as variables. Also caused invalid properties to no longer be differentiable from valid ones.

**Pull request merged**